### ISSUE TITLE
VMDK support (qemu-img convert)

### ISSUE THREAD

**Issue by sopamiria** [pbatard/rufus#1950]
Currently I am using **rufus to burn img images** **(of vmdk hard drives)** to usb drives to generate **Windows or Linux** on the go with **fully configured systems**. I also use Linux to burn those images to hard drives to perform "fast installations" with everything configured. A normal Win or Linux installation + config can take 1.5h-3h depending on the amount of software and config steps. A burn around 10min. So it saves a lot of time.

I use app like **qemu-img convert** or starwind v2v converter for the vmdk to img conversion. And later use rufus to burn the img. After every modification to the virtual machine vmdk hard drive, I need to generate the img (~30GB) before being able to burn it with rufus. It would be great to be able to **burn the vmdk directly and save the conversion step**.

Currently **rufus accepts .img.xz**, among other compressed format, and **decompresses the image on the fly**. Not sure whether it would be difficult to do something **similar for the vmdk to img on the fly** conversion step. **qemu-img convert** seems to work very well and is GPL2. The binaries are around 10MB, so it would be a question of only fetching them when that feature is used creating a tmp folder for them. 

Some frontends for **qemu-img convert** in windows for anyone finding this ticket in the while:
https://github.com/moemura/Virtual_Disk_Converter
https://github.com/maringantis/AnyConvertVM

**Comment by pbatard** [comment]
> Not sure whether it would be difficult to do something similar for the vmdk to img on the fly conversion step

It would.

> Some frontends for __qemu-img convert__

Linking frontends are pointless. What you basically need to have support for vmdk in Rufus is for _someone_ to add vmdk decompression to [Bled](https://github.com/pbatard/bled) which is what Rufus uses for all compressed disk format (and VMDK is a form of compression since it's not a flat representation of a disk image).
So the one link of interest with regards to the tools you use is a link to the actual VMDK handling of qemu, i.e. https://github.com/qemu/qemu/blob/master/block/vmdk.c and someone would need to convert that code to a `decompress_vmdk.c` one that Bled can use.

I am not planning to do that, as, to be honest, I have better things to do in the medium to long term future, especially when I expect that satisfying this request will only help you and a limited number of users who aren't exactly stuck in the first place since there exists `qemu-img.exe` and frontend tools based around that. Plus, if you're using 30 GB disk images, you're going to find that the decompression code I use in Rufus is slow (because decompression on the fly is only provided as a courtesy in Rufus, and not as a prime feature that we're going to invest a lot of time optimising), and that you are probably better of converting to dd image in terms of speed.

See also [this relevant FAQ entry](https://github.com/pbatard/rufus/wiki/FAQ#Why_cant_you_just_add_the_feature_I_want).

In short you're in a Catch 22 here, because I consider that, if not having VMDK support in Rufus bothers users so much, one of them will take it upon themselves send a patch that ads VMDK decompression to Bled, and that, as long as that hasn't happened, it must mean that it's not that big a deal...

&rarr; Closed as _"Only going to happen if __someone else__ does it"_
